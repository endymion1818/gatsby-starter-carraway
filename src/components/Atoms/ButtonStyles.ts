import { css } from 'styled-components'
import { borderradius, colors, size } from '../tokens'

const ButtonStyles = css`
  display: inline-block;
  border: none;
  padding: ${size.single} ${size.double};
  margin: 0;
  text-decoration: none;
  background: ${colors.neutral.medium};
  color: ${colors.base.primary};
  font-size: ${size.single};
  cursor: pointer;
  text-align: center;
  transition: background 250ms ease-in-out, transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: ${borderradius.medium};

  &:hover,
  &:focus {
    background: ${colors.neutral.medium};
  }

  &:focus {
    outline: 1px solid #fff;
    outline-offset: -4px;
  }

  &:active {
    transform: scale(0.99);
  }
  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
`
export default ButtonStyles
