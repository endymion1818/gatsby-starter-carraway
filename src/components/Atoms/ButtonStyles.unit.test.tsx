import { cleanup, render } from '@testing-library/react'
import React from 'react'
import styled from 'styled-components'
import ButtonStyles from './ButtonStyles'

const LooksLikeAButton = styled.a`
  ${ButtonStyles}
`

describe('ButtonStyles styles', () => {
  beforeEach(cleanup)
  it('Should render', () => {
    const { container } = render(<LooksLikeAButton href="https://ind.ie" />)
    expect(container).toMatchSnapshot()
  })
})
