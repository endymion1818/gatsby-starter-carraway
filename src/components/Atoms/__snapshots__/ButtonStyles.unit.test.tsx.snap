// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyles styles Should render 1`] = `
.c0 {
  display: inline-block;
  border: none;
  padding: 1rem 2rem;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #DCDBE2;
  color: #000080;
  font-size: 1rem;
  cursor: pointer;
  text-align: center;
  -webkit-transition: background 250ms ease-in-out,-webkit-transform 150ms ease;
  -webkit-transition: background 250ms ease-in-out,transform 150ms ease;
  transition: background 250ms ease-in-out,transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 0.5rem;
}

.c0:hover,
.c0:focus {
  background: #DCDBE2;
}

.c0:focus {
  outline: 1px solid #fff;
  outline-offset: -4px;
}

.c0:active {
  -webkit-transform: scale(0.99);
  -ms-transform: scale(0.99);
  transform: scale(0.99);
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<div>
  <a
    class="c0"
    href="https://ind.ie"
  />
</div>
`;
